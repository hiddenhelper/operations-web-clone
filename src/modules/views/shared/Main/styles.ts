import { createStyles, makeStyles } from '@material-ui/core/styles';
import { STYLE } from '../../../../constants/style';

export const useStyles = makeStyles(theme =>
  createStyles({
    appContainer: {
      width: '100%',
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
    },
    headerSpacer: {
      width: '100%',
      height: STYLE.HEADER.SPACER_HEIGHT,
    },
    sectionContainer: {
      width: '100%',
      height: 0,
      paddingLeft: 0,
      [theme.breakpoints.up('sm')]: {
        paddingLeft: STYLE.SIDEBAR.WIDTH,
      },
    },
  })
);
