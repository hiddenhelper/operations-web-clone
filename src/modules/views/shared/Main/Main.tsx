import React, { memo, useState } from 'react';

import ErrorBoundary from '../ErrorBoundary';
import Header from '../Header';
import Sidebar from '../Sidebar';
import { Logger } from '../../../services/Logger';

import { useStyles } from './styles';
import GuestHeader from '../GuestHeader';

export interface IMainProps {
  children: React.ReactNode;
  isGuest: boolean;
}

const Main = ({ children, isGuest }: IMainProps) => {
  const classes = useStyles();
  const [mobileOpen, setMobileOpen] = useState(false);

  const handleDrawerToggle = () => {
    setMobileOpen(!mobileOpen);
  };

  return (
    <>
      {isGuest ? (
        <ErrorBoundary Logger={Logger}>
          <GuestHeader />
        </ErrorBoundary>
      ) : (
        <>
          <ErrorBoundary Logger={Logger}>
            <Header handleDrawerToggle={handleDrawerToggle} />
          </ErrorBoundary>
          <ErrorBoundary Logger={Logger}>
            <Sidebar mobileOpen={mobileOpen} handleDrawerToggle={handleDrawerToggle} />
          </ErrorBoundary>
        </>
      )}
      <main
        key="Main"
        className={classes.appContainer}
        onDragOver={/* istanbul ignore next line */ e => e.preventDefault()}
        onDrop={
          /* istanbul ignore next line */ e => {
            e.stopPropagation();
            e.preventDefault();
            return false;
          }
        }
      >
        <div className={classes.headerSpacer} />
        <section className={classes.sectionContainer}>
          <ErrorBoundary Logger={Logger}>{children}</ErrorBoundary>
        </section>
      </main>
    </>
  );
};

export default memo(Main);
