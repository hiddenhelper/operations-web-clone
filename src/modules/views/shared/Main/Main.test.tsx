import React from 'react';
import { render, RenderResult } from '@testing-library/react';

import Main, { IMainProps } from './Main';

describe('Main Component', () => {
  global.console.error = () => {
    /** */
  };
  let wrapper: RenderResult;
  let props: IMainProps;

  beforeEach(() => {
    wrapper = render(
      <Main {...props}>
        <div>view</div>
      </Main>
    );
  });

  it('should render', () => {
    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
