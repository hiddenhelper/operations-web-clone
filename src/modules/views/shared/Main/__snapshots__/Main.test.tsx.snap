// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component should render 1`] = `
<body>
  <div>
    <main
      class="makeStyles-appContainer-1"
    >
      <div
        class="makeStyles-headerSpacer-2"
      />
      <section
        class="makeStyles-sectionContainer-3"
      >
        <div>
          view
        </div>
      </section>
    </main>
  </div>
</body>
`;
